.github/workflows/linux_unit_tests_with_minimum_deps.yml

```diff
@@ -0,0 +1,74 @@
+      - 'evalml/tests/dependency_update_check/**'
+  push:
+    branches:
+      - main
+
+jobs:
+  unit_tests:
+    name: ${{ matrix.python_version }} unit tests, Core Dependencies (${{ matrix.core_dependencies }})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - python_version: "3.7"
+            core_dependencies: false
+          - python_version: "3.7"
+            core_dependencies: true
+    steps:
+      - name: Set up Python ${{ matrix.python_version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python_version }}
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          fetch-depth: 2
+      - name: Update apt and install Graphviz
+        run: sudo apt update && sudo apt install -y graphviz
+      - name: Create virtual environment, upgrade pip
+        run: |
+          python -m pip install virtualenv
+          virtualenv test_python -q
+          source test_python/bin/activate
+          python -m pip install --upgrade pip -q
+      - if: ${{ matrix.core_dependencies }}
+        name: install evalml with core dependencies, & test dependencies
+        run: |
+          source test_python/bin/activate
+          python -m pip install -e . --no-dependencies
+          python -m pip install "$(cat evalml/tests/dependency_update_check/minimum_test_requirements.txt | grep wheel)"
+          python -m pip install "$(cat evalml/tests/dependency_update_check/minimum_test_requirements.txt | grep setuptools)"
+          python -m pip install -r evalml/tests/dependency_update_check/minimum_core_requirements.txt
+          python -m pip install -r evalml/tests/dependency_update_check/minimum_test_requirements.txt
+      - if: ${{ !matrix.core_dependencies }}
+        name: install evalml with core dependencies, optional requirements, & test dependencies
+        run: |
+          source test_python/bin/activate
+          python -m pip install -e . --no-dependencies
+          python -m pip install "$(cat evalml/tests/dependency_update_check/minimum_test_requirements.txt | grep wheel)"
+          python -m pip install "$(cat evalml/tests/dependency_update_check/minimum_test_requirements.txt | grep setuptools)"
+          python -m pip install -r evalml/tests/dependency_update_check/minimum_core_requirements.txt
+          python -m pip install -r evalml/tests/dependency_update_check/minimum_requirements.txt
+          python -m pip install -r evalml/tests/dependency_update_check/minimum_test_requirements.txt
+      - if: ${{ matrix.core_dependencies }}
+        name: run tests with minimal dependencies
+        run: |
+          source test_python/bin/activate
+          python -m pytest evalml/ -x -n 2 --doctest-modules --has-minimal-dependencies
```

Comment:
Not a big deal but feel free to add this and the command below to the makefile for easier reference.